88 covariate visualisation

edarin opened the pull request:
The covariates visualisation is stored in src/covariates/report.
I have organised it into three different reports, each consisting of a Quarto .qmd file and its rendered .html version.

The three reports are as follows:

1.  1_covariates_vis: _Visualise individual covariates_. This report contains scatterplots over time for each covariate along with their 31 + 2 × 3 related standardisations:
-  Raw values
-  Cumulative values (with three different time windows)
-  Time-scaled values (with three different time windows)

2. 2_covariates_corr: _Correlation between covariates_. This report provides a correlation analysis split into three sections:
-  Correlation between processing methods for each covariate
-  Correlation between covariates for each processing method
-  A full correlation table for further exploration

3. 3_covariates_popChange: _Correlation with population change_. This is my favourite. It analyses the correlation between covariates and a proxy for population change (daily changes in deterministic population estimates). The analysis distinguishes between population increase and decrease, evaluating correlations based on:
-  Current value of the covariate
-  Lagged values (up to -5 weeks)
-  Lead values (up to +5 weeks)

edarin committed:
add macro region

edarin committed:
add deterministic estimates to repo

edarin committed:
visualise individual covariates

edarin committed:
visualise correlation between covariates

edarin committed:
visualise correlation with pop change proxy from deterministic estimated

edarin committed:
add html dependencies

edarin committed:
add packages needed for vis

edarin committed:
remove collasped rows

alaparicioc commented:
Hi, 

Some comments related to report 1 (1_covariates_vis: Visualise individual covariates.),
* could you clarify to me from where you are taken the values 31 + 2 × 3 related standardisations?
* could some summary statistics be provided? For example, in the case of war fires, it is difficult to see the max of the variable for the South.
* Could the nature of the covariates be provided? Are they categorical, discrete, continuous?
* Could a count of total missing values and level of missingness be provided e.g. by oblast and time?
*  why some covariates are flat? There is not variation across the entire period (e.g. sirens in all macroregions/oblasts but for Khmeknytska and City)?
* Did you smooth to any extend any covariate? e.g. pwtt. 
* In the occupied and occupied_prop, there are small variations e.g. in the North, but the flat lines return to the exact value after some time?

Further comments will continue below

alaparicioc commented on src/1_pop_data/50_combine_deterministicDeliverables.R:
```diff
@@ -0,0 +1,64 @@
+  select(ADM1_PCODE, collection_date, starts_with('m'),total,  starts_with('f')) |>
+  left_join(time_index) |> 
+  group_by(ADM1_PCODE, t) |> 
+  summarise(across(c(starts_with('m'), starts_with('f'), total), ~mean(.x, na.rm = TRUE))) |> 
```
why are you getting the mean of the population here? At the beginning, I thought that you were averaging the daily population every 7 days for our weekly estimates, but the collection dates seem weekly. So, I do not get why this.

doug-leasure commented:
Super cool!! Thanks for doing this. I really like the correlation with pop change across different windows. 

I had some of the same questions as @alaparicioc , like what's going on with the sirens data being flat line everywhere? 

I would also suggest allowing the plots to be visualised with the raw data values rather than only the standardised values as this would help answer some of @alaparicioc 's questions about data types, etc. 

Once @alaparicioc approves the PR, please go ahead and merge without waiting for my approval.

alaparicioc commented:
in terms of the 2_covariates_corr: Correlation between covariates, it is interesting to see no correlation between the sum and the raw data in some covariates. I was expecting some correlation between different ways of processing the same covariate, suggesting some consistency in the patterns detected across methods. In this case, covariates with high correlations across different methods could serve as reliable indicators for modelling. For unexpected correlations might require re-evaluation of data preprocessing.

alaparicioc commented:
Related to 3_covariates_popChange: Correlation with population change, I agree with you this report can provide the most valuable insights. I have some questions/comments:

* Could you clarify to me the future week column? I do not get it conceptually.
* If "correlation can vary between -100 (dark blue) and 100 (dark red)", why are there dark blues with negative values? (see e.g. pwtt 12week, 24week, 4week in cumulate covariate)
* Could you guide me to your chain of thoughts, why did you decide on splitting the df in 
![image](https://github.com/user-attachments/assets/4243adfd-bd67-4fbd-9ba4-4ddc02695932)?
I do not get it, and it is difficult for me to comment on the values and insights this analysis can provide.

For Covariates correlated with positive population change,
* Current values of some variables seem to be the strongest
* Apart from current, the previous week values shows the highest numbers, which is interesting since our weekly time step would not be that bad to produce useful estimates 
* It is nice to see expected relatively strong correlation between population change and acled_eventAgainstCivilians and acled_eventBattle.
* The results of scalin covariates are not clear to me.

edarin commented:
>  could you clarify to me from where you are taken the values 31 + 2 × 3 related standardisations?

This a typo. It is 17 (raw) + 17 * 2 (mean, sum) * 3 (windows) = 119

> could some summary statistics be provided? For example, in the case of war fires, it is difficult to see the max of the variable for the South.

Sure. Dont hesitate to change the code if needed as well.
    
> Could the nature of the covariates be provided? Are they categorical, discrete, continuous?

Added to the first table

> Could a count of total missing values and level of missingness be provided e.g. by oblast and time?

 What do you call missingness? We are only able to detect two sources of missingness: (1) due to spatial coverage of the datasets (e.g. sirens dataset doesnt include Crimea) and (2) due to temporal coverage (pwtt is monthly). I have added a coverage column in Table 1. 
 

> why some covariates are flat? There is not variation across the entire period (e.g. sirens in all macroregions/oblasts but for Khmeknytska and City)?

Thanks. There was an issue in the processing. It is now corrected.

> Did you smooth to any extend any covariate? e.g. pwtt.

I followed what I said in previous issuw. Pwtt replicates the same value until a new change that is for 4 time points. And obviously all standardisation brings some form of smoothness
    

> In the occupied and occupied_prop, there are small variations e.g. in the North, but the flat lines return to the exact value after some time?

They has been no invasion, except at the very beginning for Sumska.

> why are you getting the mean of the population here? At the beginning, I thought that you were averaging the daily population every 7 days for our weekly estimates, but the collection dates seem weekly. So, I do not get why this.

The deterministic estimates have been produced daily. I am averaging over a week to mimic the Bayesian observation model such that it matches t which has a weekly timestep.

> I would also suggest allowing the plots to be visualised with the raw data values rather than only the standardised values as this would help answer some of @alaparicioc 's questions about data types, etc.

The raw data is visualised as part of the plot. See category "raw, NA, NA"

> it is interesting to see no correlation between the sum and the raw data in some covariates. I was expecting some correlation between different ways of processing the same covariate, suggesting some consistency in the patterns detected across methods. In this case, covariates with high correlations across different methods could serve as reliable indicators for modelling. For unexpected correlations might require re-evaluation of data preprocessing.

Interesting take. I would have said that if the processing makes the data uncorrelated it's more that the processing is interesting rather than the data is unreliable. Your shout!

>  Could you clarify to me the future week column? I do not get it conceptually.
These are the correlation of pop_change_t with cov_{t+1), cov_{t+2} etc

>  If "correlation can vary between -100 (dark blue) and 100 (dark red)", why are there dark blues with negative values? (see e.g. pwtt 12week, 24week, 4week in cumulate covariate)

Because darks blue depicts a correlation from -100 to 0 with darker shades closer to -100.

> Could you guide me to your chain of thoughts, why did you decide on splitting the df in
  image?  I do not get it, and it is difficult for me to comment on the values and insights this analysis can provide.

We were discussing that with Ridhi, Doug and you maybe 1000 months ago. The idea is to separate drivers of population increase vs drivers of population decrease, that is push and pull factors. For example an airstrike could have an effect on negative pop change but none on positive pop change. Do tell me if it doesnt make sense. In the end in the model you wont integrate the difference I believe so maybe adding a table that have the correlation with pop_change in general might be wise.

> The results of scalin covariates are not clear to me.

It is not indeed striking. However it is interesting for example that `acled_eventProtest` in its cumulative form has a strong negative effect on population decrease, and in its temporal scaling form it has a strong positive effect. It means that continuously high number of protest reduces the population decrease but if the current protestEvent is stronger than the past ones it increases the population decrease.

edarin committed:
fix i_name mapping in sirens

edarin committed:
update report

alaparicioc commented:
Sorry for my delayed response. You can find my quick responses below:

Got it!
> > could you clarify to me from where you are taken the values 31 + 2 × 3 related standardisations?
> 
> This a typo. It is 17 (raw) + 17 * 2 (mean, sum) * 3 (windows) = 119

Brilliant! I may include them but not this week.
> > could some summary statistics be provided? For example, in the case of war fires, it is difficult to see the max of the variable for the South.
> 
> Sure. Dont hesitate to change the code if needed as well.

Thanks! 
> > Could the nature of the covariates be provided? Are they categorical, discrete, continuous?
> 
> Added to the first table

Oh, I see. Thanks for adding coverage column.
> > Could a count of total missing values and level of missingness be provided e.g. by oblast and time?
> 
> What do you call missingness? We are only able to detect two sources of missingness: (1) due to spatial coverage of the datasets (e.g. sirens dataset doesnt include Crimea) and (2) due to temporal coverage (pwtt is monthly). I have added a coverage column in Table 1.

Brilliant!
> > why some covariates are flat? There is not variation across the entire period (e.g. sirens in all macroregions/oblasts but for Khmeknytska and City)?
> 
> Thanks. There was an issue in the processing. It is now corrected.

This question was because I expected that continous line looked different due to the repetition of the value until a new one come up. I was expecting more something similar to stepwise lines.
> > Did you smooth to any extend any covariate? e.g. pwtt.
> 
> I followed what I said in previous issuw. Pwtt replicates the same value until a new change that is for 4 time points. And obviously all standardisation brings some form of smoothness

Yes, but they return to the same level of occupied territory? I was expecting a flat line when occupied territories did not change, but not to the same level/figure. 
> > In the occupied and occupied_prop, there are small variations e.g. in the North, but the flat lines return to the exact value after some time?
> 
> They has been no invasion, except at the very beginning for Sumska.


Yeah, you are right.
> > why are you getting the mean of the population here? At the beginning, I thought that you were averaging the daily population every 7 days for our weekly estimates, but the collection dates seem weekly. So, I do not get why this.
> 
> The deterministic estimates have been produced daily. I am averaging over a week to mimic the Bayesian observation model such that it matches t which has a weekly timestep.



Thanks for the clarification!
> > Could you clarify to me the future week column? I do not get it conceptually.
> > These are the correlation of pop_change_t with cov_{t+1), cov_{t+2} etc

Yes, you are right. I got colours mixed! My questions was referring to reddish cells with negative values. I will blame my sight on this one. The colour scale can be fine, it is just I am not seein it very well.  
> > If "correlation can vary between -100 (dark blue) and 100 (dark red)", why are there dark blues with negative values? (see e.g. pwtt 12week, 24week, 4week in cumulate covariate)
> 
> Because darks blue depicts a correlation from -100 to 0 with darker shades closer to -100.


Yeah, I do not think that the model will split the data like this. But I suppose this could help to interpret the results. 
> > Could you guide me to your chain of thoughts, why did you decide on splitting the df in
> > image?  I do not get it, and it is difficult for me to comment on the values and insights this analysis can provide.
> 
> We were discussing that with Ridhi, Doug and you maybe 1000 months ago. The idea is to separate drivers of population increase vs drivers of population decrease, that is push and pull factors. For example an airstrike could have an effect on negative pop change but none on positive pop change. Do tell me if it doesnt make sense. In the end in the model you wont integrate the difference I believe so maybe adding a table that have the correlation with pop_change in general might be wise.

doug-leasure commented:
This is really nice work. Merge, merge, merge! ;) 

One note:  The sirens dataset documentation says that there have been constant sirens in Luhansk and Crimea, so they do not include them in the dataset. We want to be sure that this is reflected in our covariates values (i.e. missing data here is not zero... it is maximum sirens all the time).

doug-leasure commented:
Another note: It seems that covariates shown are all scaled relative to oblast-specific means rather than global means across all oblasts and time periods. The scaled values always have a mean of zero for every oblast. This would be quite detrimental to model performance because it hides the signal of differences between oblasts. Is this because of the specific data you've chosen to show in `1_covariates_vis` or is this a broader pattern across all of the standardised covariates?

edarin commented:
> Another note: It seems that covariates shown are all scaled relative to oblast-specific means rather than global means across all oblasts and time periods. The scaled values always have a mean of zero for every oblast. This would be quite detrimental to model performance because it hides the signal of differences between oblasts. Is this because of the specific data you've chosen to show in 1_covariates_vis or is this a broader pattern across all of the standardised covariates?

It looks like it because the values are tiny but in practice it is not centered per oblast-specific mean:

![image](https://github.com/user-attachments/assets/4ca87657-f891-4929-a37b-5ce5a9902348)

(the mean of the centered cov is not exactly 0 because some centering values are custom and not the exact mean of the cov)

edarin committed:
Merge branch 'dev' into 88-covariate-visualisation

edarin commented:
> This is really nice work. Merge, merge, merge! ;)
> 
> One note: The sirens dataset documentation says that there have been constant sirens in Luhansk and Crimea, so they do not include them in the dataset. We want to be sure that this is reflected in our covariates values (i.e. missing data here is not zero... it is maximum sirens all the time).

It is said: 
    In Luhansk region from April 4 at 04:45 PM (UTC+00) or April 4 at 07:45 PM (local time)
    In Crimea from December 10 at 10:22 PM (UTC+00) or December 11 at 12:22 AM (local time)

If I follow the exact date given this will create a jump in the middle of the time series. Shall I do from the beginning of the war? @doug-leasure

doug-leasure commented:
> If I follow the exact date given this will create a jump in the middle of the time series. Shall I do from the beginning of the war? @doug-leasure

@edarin I don't have a strong opinion on this right now. I would need to look at the time series to see what the trend/jump is like. Please use your best judgement here, and then we can revisit if needed once we see how the covariate performs in the model.

edarin committed:
fix permanent siren in luhansk and crimea

edarin committed:
Merge branch 'dev' into 88-covariate-visualisation
